alacritty/src/input/keyboard.rs

```diff
@@ -107,15 +107,22 @@ impl<T: EventListener, A: ActionContext<T>> Processor<T, A> {
         mods: ModifiersState,
     ) -> bool {
         if mode.contains(TermMode::REPORT_ALL_KEYS_AS_ESC) {
-            true
-        } else if mode.contains(TermMode::DISAMBIGUATE_ESC_CODES) {
+            return true;
+        }
+
+        let disambiguate = if mode.contains(TermMode::DISAMBIGUATE_ESC_CODES) {
             let on_numpad = key.location == KeyLocation::Numpad;
             let is_escape = key.logical_key == Key::Named(NamedKey::Escape);
             is_escape || (!mods.is_empty() && mods != ModifiersState::SHIFT) || on_numpad
         } else {
-            // `Delete` key always has text attached to it, but it's a named key, thus needs to be
-            // excluded here as well.
-            text.is_empty() || key.logical_key == Key::Named(NamedKey::Delete)
+            false
+        };
+
+        match key.logical_key {
+            _ if disambiguate => true,
```

Comment:
Not sure this has any business being inside this match statement. Though I guess it's more concise than the 3-4 lines for a separate conditional.

Reply:
yeah, I just don't like `if true { true }` branch.